src/clojure/nrepl/middleware/interruptible_eval.clj

```diff
@@ -76,7 +76,13 @@
                     (push-thread-bindings bindings))
            :read (if (string? code)
                    (let [reader (source-logging-pushback-reader code line column)]
-                     #(read {:read-cond :allow :eof %2} reader))
+                     #(try (read {:read-cond :allow :eof %2} reader)
+                           (catch RuntimeException e
+                             ;; If error happens during reading the string, we
+                             ;; don't want eval to start reading and executing the
+                             ;; rest of it. So we skip over the remaining text.
+                             (while (pos? (.skip ^LineNumberingPushbackReader reader 10000)))
```

Comment:
Would calling `.close` on the reader work rather than skipping an arbitrary number of characters?

Reply:
It doesn't skip arbitrary number of characters - it skips everything left in the reader. If you refer to `10000`, that's just a "skipping step" since `.skip` doesn't have an API to "skip everything".

Calling `read` on a closed reader throws an exception, so it's not the cleanest way to do things. Skipping over the remaining content seems to me like the most natural thing to do, so that EOF logic remains unchanged.

Reply:
You're right, `.close` is no good. Maybe `(.skip ^LineNumberingPushbackReader reader Long/MAX_VALUE)` without the loop and check?

Reply:
Yeah, I considered that too but went with a loop. I guess that might be better, let me change it.